skimage/util/crop.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+
+__all__ = ['crop']
+
+
+def crop(image, bounding_box, axis=None):
+    """Crop an image from a bounding box.
+        Bounding_box (which is a 2-tuple (min_val, max_val) for each axis)
+        and (optional) axis for corresponding axis order to bounding_box.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input array.
+    bounding_box : list of 2-tuple (x, y) where x < y.
+        Bounding box.
+    axis : tuple, optional
+        Axis order for cropping.
+        if provided, same legth as bounding_box.
+
+
+    Returns
+    ----------
+    out : ndarray
+        Cropped array.
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> from skimage.util.crop import crop
+    >>> img = data.camera()
+    >>> img.shape
+    (512, 512)
+    >>> cropped_img = crop(img, [(0, 100)])
+    >>> cropped_img.shape
+    (100, 512)
+    >>> cropped_img = crop(img, [(0, 100), (0, 100)])
+    >>> cropped_img.shape
+    (100, 100)
+    >>> cropped_img = crop(img, [(0, 100), (0, 75)], axis=[1, 0])
+    >>> cropped_img.shape
+    (75, 100)
+    """
+
+    # empty length of bounding box detected on None detected
+    if not bounding_box:
+        return image
+
+    # check data isinstance of numpy array
+    if not isinstance(image, np.ndarray):
+        raise ValueError("data must be numpy array")
+
+    # if not axis provided,
+    # consider sequential cropping on axis
+    if not axis:
+        axis = list(range(len(bounding_box)))
```

Comment:
We should decide if using `axes=None` should use the first or the last `len(bounding_box)` axes. For examples from other libraries, NumPy functions tend to default to `axis=-1` and PyWavelets 2D transforms default to `axes=(-2, -1)`. I don't have a very strong preference either way as long as the choice is documented in the description of `axes` in the docstring parameters.

Reply:
updated documentation for default choice.